### DISCUSSION TITLE
DecryptException thrown when spam bot tries to add a value to the valid from field

### DISCUSSION THREAD

**Discussion by jayan-blutui** [spatie/laravel-honeypot#108] [Q&A] [answered]
Hello,

I came across an issue with the `valid_from_field_name` and/or `valid_from_timestamp`, where a spam bot could change the value from an encrypted timestamp into a string such as `Hobbiton` (Didn't want to mention any particular country). Changing the encrypted string to any other value would then cause an error in the `EncryptedTime` class, resulting in a `DecryptException`, obviously because the changed value isn't encrypted.

I would suggest converting line 27 in the `EncryptedTime` class to a `try/catch` and possibly return an `InvalidTimestamp` exception? The line to change: 

```php
$timestamp = app('encrypter')->decrypt($encryptedTime);
```

Unsure if this is something I can make a pull request for or would it be a breaking change? Happy to get anyone's comments on this.

**Comment by freekmurze** [comment, marked as answer]
I'd welcome a PR that fixes this. If possible, add a test 👍

**Comment by mikemunger** [comment]
Having the same issue in a production system. I took a crack at it with a pull request.

https://github.com/spatie/laravel-honeypot/pull/110

**Reply by jayan-blutui** [reply]
This looks like the best solution IMO. I actually haven't had the chance to look into this, but I think I would have done it the same way.
